packages/@react-aria/utils/src/useUpdateEffect.ts

```diff
@@ -25,10 +25,11 @@ export function useUpdateEffect(effect: EffectCallback, dependencies: any[]): ((
   }, []);
 
   useEffect(() => {
+    let prevDeps = lastDeps.current;
     if (isInitialMount.current) {
       isInitialMount.current = false;
-    } else if (!lastDeps.current || dependencies.some((dep, i) => !Object.is(dep, lastDeps[i]))) {
-      return effect();
+    } else if (!prevDeps || dependencies.some((dep, i) => !Object.is(dep, prevDeps[i]))) {
+      effect();
     }
```

Comment:
This hook seemed to be entirely broken? `lastDeps[i]` never exists (should be `lastDeps.current[I]`). And the early return prevented `lastDeps.current` from being updated.